### PR TITLE
EN-7818 hardfork resolver fixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed the resolvers creation in the hardfork process as to request the data on proper topics

- added more prints regarding node status when doing the export process

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: update/factory/fullSyncResolversContainerFactory.go [hunk 1]
```diff
@@ -148,11 +144,19 @@ func (rcf *resolversContainerFactory) checkIfResolverExists(topic string) bool {
...
-func (rcf *resolversContainerFactory) createTrieNodesResolver(baseTopic string, trieId string) (dataRetriever.Resolver, error) {
+func (rcf *resolversContainerFactory) createTrieNodesResolver(baseTopic string, trieId string, targetShardID uint32) (dataRetriever.Resolver, error) {
+	//for each resolver we create a pseudo-intra shard topic as to make at least of half of the requests target the proper peers
+	//this pseud-intra shard topic is the consensus_targetShardID
+	targetShardCoordinator, err := sharding.NewMultiShardCoordinator(rcf.shardCoordinator.NumberOfShards(), targetShardID)
+	if err != nil {
+		return nil, err
+	}
+
+	targetConsensuStopic := core.ConsensusTopic + targetShardCoordinator.CommunicationIdentifier(targetShardID)
```
Comment (SebastianMarian) [inline update/factory/fullSyncResolversContainerFactory.go:155]: Consensus

File: update/sync/coordinator.go [hunk 2]
```diff
@@ -61,23 +64,31 @@ func NewSyncState(args ArgsNewSyncState) (*syncState, error) {
...
 	ss.syncingEpoch = epoch
 	err := ss.headers.SyncUnFinishedMetaHeaders(epoch)
+	cancelDisplay()
 	if err != nil {
-		return err
+		return fmt.Errorf("%w in syncState.SyncAllState - SyncUnFinishedMetaHeaders", err)
 	}
 
+	ctxDisplay, cancelDisplay = context.WithCancel(context.Background())
+	go displayStatusMessage("syncing epoch start metablock", ctxDisplay)
```
Comment (SebastianMarian) [inline update/sync/coordinator.go:77]: You also can add epoch like in the one from line 68

Comment (SebastianMarian) [inline update/sync/coordinator.go:77]: getting

File: update/sync/coordinator.go [hunk 3]
```diff
@@ -61,23 +64,31 @@ func NewSyncState(args ArgsNewSyncState) (*syncState, error) {
...
+	cancelDisplay()
 	if err != nil {
-		return err
+		return fmt.Errorf("%w in syncState.SyncAllState - GetEpochStartMetaBlock", err)
 	}
 
 	ss.printMetablockInfo(meta)
 
+	ctxDisplay, cancelDisplay = context.WithCancel(context.Background())
+	go displayStatusMessage("syncing un-finished metablocks", ctxDisplay)
```
Comment (SebastianMarian) [inline update/sync/coordinator.go:87]: getting, syncing is in line 70

File: update/sync/coordinator.go [hunk 4]
```diff
@@ -101,49 +112,65 @@ func (ss *syncState) SyncAllState(epoch uint32) error {
...
 		if errSync != nil {
 			mutErr.Lock()
-			errFound = errSync
+			errFound = fmt.Errorf("%w in syncState.SyncAllState - SyncPendingMiniBlocksFromMeta", errSync)
 			mutErr.Unlock()
 			return
 		}
 
+		ctxDisplay, cancelDisplay = context.WithCancel(context.Background())
+		go displayStatusMessage("syncing miniblocks", ctxDisplay)
```
Comment (SebastianMarian) [inline update/sync/coordinator.go:130]: getting

File: update/sync/coordinator.go [hunk 5]
```diff
@@ -101,49 +112,65 @@ func (ss *syncState) SyncAllState(epoch uint32) error {
...
 			mutErr.Unlock()
 			return
 		}
 
+		ctxDisplay, cancelDisplay = context.WithCancel(context.Background())
+		go displayStatusMessage(fmt.Sprintf("syncing pending transactions for epoch %d", epoch), ctxDisplay)
 		ctx, cancel = context.WithTimeout(context.Background(), time.Hour)
 		errSync = ss.transactions.SyncPendingTransactionsFor(syncedMiniBlocks, ss.syncingEpoch, ctx)
 		cancel()
+		cancelDisplay()
```
Comment (SebastianMarian) [inline update/sync/coordinator.go:145]: cancelDisplay is done before cancel in line 120-121. Does it matter?

Comment (SebastianMarian) [reply update/sync/coordinator.go:145]: Not each Sync... calls from this method has cancel method called afterwards

File: update/factory/fullSyncResolversContainerFactory.go [hunk 6]
```diff
@@ -148,11 +144,19 @@ func (rcf *resolversContainerFactory) checkIfResolverExists(topic string) bool {
 	return err == nil
 }
 
-func (rcf *resolversContainerFactory) createTrieNodesResolver(baseTopic string, trieId string) (dataRetriever.Resolver, error) {
+func (rcf *resolversContainerFactory) createTrieNodesResolver(baseTopic string, trieId string, targetShardID uint32) (dataRetriever.Resolver, error) {
+	//for each resolver we create a pseudo-intra shard topic as to make at least of half of the requests target the proper peers
+	//this pseud-intra shard topic is the consensus_targetShardID
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply update/factory/fullSyncResolversContainerFactory.go:149]: done

File: update/factory/fullSyncResolversContainerFactory.go (hunk 1 above)
Comment (iulianpascalau) [reply update/factory/fullSyncResolversContainerFactory.go:155]: done

File: update/sync/coordinator.go (hunk 2 above)
Comment (iulianpascalau) [reply update/sync/coordinator.go:77]: done

Comment (iulianpascalau) [reply update/sync/coordinator.go:77]: done

File: update/sync/coordinator.go (hunk 3 above)
Comment (iulianpascalau) [reply update/sync/coordinator.go:87]: done

File: update/sync/coordinator.go (hunk 4 above)
Comment (iulianpascalau) [reply update/sync/coordinator.go:130]: done

File: update/sync/coordinator.go (hunk 5 above)
Comment (iulianpascalau) [reply update/sync/coordinator.go:145]: refactored the order. However, the missing cancel() comes from the fact that the `GetMiniBlocks` function does not require a context

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**